src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/unit/restapi/controller/ReferralControllerTest.kt

```diff
@@ -652,4 +653,219 @@ constructor(
+
+    paginatedReferralSummary.pageSize shouldBe 10
+    paginatedReferralSummary.totalPages shouldBe 1
+    paginatedReferralSummary.content?.shouldHaveSize(2)
+    paginatedReferralSummary.totalElements shouldBe 2L
+
+    val firstReferral = paginatedReferralSummary.content?.find { it.id == firstReferralId }
+    firstReferral shouldNotBe null
+    firstReferral?.let { referral ->
+      referral.courseName shouldBe "Course for referralSummary1"
+      referral.audiences shouldContainExactlyInAnyOrder audiencesForFirstReferral
+      referral.status shouldBe ReferralStatus.referralStarted
+      referral.prisonNumber shouldBe PRISON_NUMBER
+      referral.referrerUsername shouldBe CLIENT_USERNAME
+    }
+
+    val secondReferral = paginatedReferralSummary.content?.find { it.id == secondReferralId }
+    secondReferral shouldNotBe null
+    secondReferral?.let { referral ->
+      referral.courseName shouldBe "Course for referralSummary2"
+      referral.audiences shouldContainExactlyInAnyOrder audiencesForSecondReferral
+      referral.status shouldBe ReferralStatus.referralSubmitted
+      referral.submittedOn shouldBe LocalDateTime.MIN.toString()
+      referral.prisonNumber shouldBe PRISON_NUMBER
+      referral.referrerUsername shouldBe CLIENT_USERNAME
+    }
+
+    verify { referralService.getReferralsByUsername(CLIENT_USERNAME, pageable, null, null, null) }
+  }
+
+  @Test
+  fun `getReferralsByUsername with valid username and custom pagination count will return 200 with paginated body for each page`() {
+    val pageSize = 1
+    val pageableFirstPage = PageRequest.of(0, pageSize, Sort.by("referralId"))
+    val pageableSecondPage = PageRequest.of(1, pageSize, Sort.by("referralId"))
+
+    val firstReferralId = UUID.randomUUID()
+    val audiencesForFirstReferral = listOf("Audience 1", "Audience 2", "Audience 3")
+    audiencesForFirstReferral.map { audience ->
+      ReferralSummaryProjectionFactory()
+        .withReferralId(firstReferralId)
+        .withCourseName("Course name")
+        .withAudience(audience)
+        .withStatus(ReferralEntity.ReferralStatus.REFERRAL_STARTED)
+        .withPrisonNumber(PRISON_NUMBER)
+        .withReferrerUsername(REFERRER_USERNAME)
+        .produce()
+    }
+
+    val secondReferralId = UUID.randomUUID()
+    val audiencesForSecondReferral = listOf("Audience 4", "Audience 5", "Audience 6")
+    audiencesForSecondReferral.map { audience ->
+      ReferralSummaryProjectionFactory()
+        .withReferralId(secondReferralId)
+        .withCourseName("Another course name")
+        .withAudience(audience)
+        .withStatus(ReferralEntity.ReferralStatus.REFERRAL_SUBMITTED)
+        .withSubmittedOn(LocalDateTime.MIN)
+        .withPrisonNumber(PRISON_NUMBER)
+        .withReferrerUsername(REFERRER_USERNAME)
+        .produce()
+    }
+
+    every { referralService.getReferralsByUsername(CLIENT_USERNAME, pageableFirstPage, null, null, null) } returns
```

Comment:
Should this be `REFERRER_USERNAME`, and elsewhere in the test below or am I missing something?